R/opts.R

```diff
@@ -917,6 +917,299 @@ opt_table_font <- function(
+      data_hlines_color = preset_colors$gray[[3]],
+      data_vlines_style = "none",
+      data_vlines_color = preset_colors$gray[[3]],
+      summary_row_background_color = "#FFFFFF",
+      grand_summary_row_background_color = preset_colors$gray[[2]],
+      row_striping_background_color = preset_colors$gray[[1]],
+      table_outline_color = NA_character_
+    )
+  ) %>%
+  dplyr::bind_rows(
+    dplyr::tibble( # style 4
+      color = names(preset_colors),
+      style = 4,
+      table_hlines_color = preset_colors$gray[[2]],
+      location_hlines_color = preset_colors$gray[[2]],
+      column_labels_background_color = get_all_preset_colors_at_index(4),
+      stub_background_color = preset_colors$gray[[3]],
+      stub_border_style = "dashed",
+      stub_border_color = preset_colors$gray[[2]],
+      data_hlines_style = "dashed",
+      data_hlines_color = preset_colors$gray[[2]],
+      data_vlines_style = "dashed",
+      data_vlines_color = preset_colors$gray[[2]],
+      summary_row_background_color = "#FFFFFF",
+      grand_summary_row_background_color = get_all_preset_colors_at_index(4),
+      row_striping_background_color = preset_colors$gray[[1]],
+      table_outline_color = preset_colors$gray[[2]]
+    )
+  ) %>%
+  dplyr::bind_rows(
+    dplyr::tibble( # style 5
+      color = names(preset_colors),
+      style = 5,
+      table_hlines_color = preset_colors$gray[[2]],
+      location_hlines_color = preset_colors$gray[[2]],
+      column_labels_background_color = get_all_preset_colors_at_index(5),
+      stub_background_color = preset_colors$gray[[3]],
+      stub_border_style = "solid",
+      stub_border_color = preset_colors$gray[[2]],
+      data_hlines_style = "solid",
+      data_hlines_color = preset_colors$gray[[2]],
+      data_vlines_style = "solid",
+      data_vlines_color = preset_colors$gray[[2]],
+      summary_row_background_color = preset_colors$gray[[4]],
+      grand_summary_row_background_color = preset_colors$gray[[3]],
+      row_striping_background_color = preset_colors$gray[[1]],
+      table_outline_color = preset_colors$gray[[2]]
+    )
+  ) %>%
+  dplyr::bind_rows(
+    dplyr::tibble( # style 6
+      color = names(preset_colors),
+      style = 6,
+      table_hlines_color = preset_colors$gray[[4]],
+      location_hlines_color = preset_colors$gray[[4]],
+      column_labels_background_color = get_all_preset_colors_at_index(4),
+      stub_background_color = get_all_preset_colors_at_index(2),
+      stub_border_style = "solid",
+      stub_border_color = preset_colors$gray[[2]],
+      data_hlines_style = "none",
+      data_hlines_color = preset_colors$gray[[2]],
+      data_vlines_style = "none",
+      data_vlines_color = preset_colors$gray[[2]],
+      summary_row_background_color = "#FFFFFF",
+      grand_summary_row_background_color = preset_colors$gray[[2]],
+      row_striping_background_color = get_all_preset_colors_at_index(1),
+      table_outline_color = NA_character_
+    )
+  )
```

Comment:
## general design

It's possible I'm totally missing something, but I'm wondering about this design:

- where did these styles come from?
- can they be extended?

## performance

In addition, is this a sequence of `bind_rows` each creating a single-row tibble? This predetermined table gets initialized every time we call `library(gt)`, is that right? If so, does it make sense to have a script during the build step of the package that creates a .csv file or something like it and then we access that directly at runtime when needed? I'm sure this one initialization step won't take a lot of time by itself, but add a couple of dozen of these and we're now talking about maybe tenth of a second of loading time for every single `gt` user out there, regardless of whether they use this or not.

Reply:
Styles are all in a Sketch doc I put together quite a while back. With this iteration they can't be extended by the user but I'd like to make that possible in a follow-up PR.